### PR TITLE
[HFX-1841] CP-17631: Don't use a default QEMU keymap

### PR DESCRIPTION
**Pull request by sharady**

### CODE REVIEW

**Comment by sharady** [comment]
Needs review, Commit https://github.com/xapi-project/xenopsd/pull/272/commits/97f3c6d785ee3874a906615757ae295880e393ee as part of https://github.com/xapi-project/xenopsd/pull/272 not needed on cream.

File: ocaml/xenops/device.ml [hunk 1]
```diff
@@ -1775,27 +1775,31 @@ let cmdline_of_disp info =
 	in
 	let videoram_opt = ["-videoram"; string_of_int info.video_mib] in
 	let dom0_input_opts x =
-	  (maybe_with_default [] (fun i -> ["-dom0-input"; string_of_int i]) x)
+		(maybe_with_default [] (fun i -> ["-dom0-input"; string_of_int i]) x)
+	in
+	let vnc_opts_of ip_addr_opt auto port keymap =
```
Comment (lindig) [inline ocaml/xenops/device.ml:1780]: The naming of values is a bit inconsistent: the IP address and the keymap are both optional values but one has `opt` in its name and the other doesn't. 

Independent of that I've tried to come up with more regular code but haven't found something that is vastly better.

**Review by lindig — APPROVED** [review]
I made a comment about naming. If this is a backport, don't bother changing it. If this is new code, please consider changing the names.

File: ocaml/xenops/device.ml (hunk 1 above)
Comment (sharady) [reply ocaml/xenops/device.ml:1780]: Since this is a backport, lets keep the commits and commit history intact as we have in trunk.

**Comment by robhoes** [comment]
@lindig When you reviewed this, did you check that the code was backported correctly?

**Comment by lindig** [comment]
I did not since I assumed these are cherry-picks. Will take a closer look.

**Comment by robhoes** [comment]
@lindig Thanks. Backports from post-Cream are slightly complicated cherry-picks, because repos and paths are changed. So it's good to verify this.

**Comment by lindig** [comment]
Surprisingly, the original fix was in xenopds whereas these commits are in xapi. I believe the backport is faithful to the original.

**Pull request merged**